frontend/src/lib/components/PropertyFilters/components/TaxonomicPropertyFilter.tsx

```diff
@@ -24,9 +28,28 @@ export function TaxonomicPropertyFilter({
     taxonomicGroupTypes,
 }: PropertyFilterInternalProps): JSX.Element {
     const pageKey = useMemo(() => pageKeyInput || `filter-${uniqueMemoizedIndex++}`, [pageKeyInput])
+    const groupTypes = taxonomicGroupTypes || [
+        TaxonomicFilterGroupType.EventProperties,
+        TaxonomicFilterGroupType.PersonProperties,
+        TaxonomicFilterGroupType.Cohorts,
+        TaxonomicFilterGroupType.Elements,
+    ]
+    const taxonomicOnChange: (group: TaxonomicFilterGroup, value: TaxonomicFilterValue, item: any) => void = (
+        taxonomicGroup,
+        value
+    ) => {
+        selectItem(taxonomicGroup, value)
+        if (taxonomicGroup.type === TaxonomicFilterGroupType.Cohorts) {
+            onComplete?.()
+        }
+    }
     const { setFilter } = useActions(propertyFilterLogic)
-
-    const logic = taxonomicPropertyFilterLogic({ pageKey, filterIndex: index })
+    const logic = taxonomicPropertyFilterLogic({
+        pageKey,
+        filterIndex: index,
+        taxonomicGroupTypes: groupTypes,
+        taxonomicOnChange,
+    })
```

Comment:
Had to move things around in this file since this logic was connecting to `taxonomicFilterLogic`, but not passing any props.